src/Product/Unit/Create.php

```diff
@@ -63,7 +63,7 @@ public function create(Unit $unit)
 			'visible'	=> (bool) $unit->visible,
 			'barcode'	=> $unit->barcode,
 			'sup_ref'	=> $unit->supplierRef,
-			'weight'	=> $unit->weight,
+			'weight'	=> $unit->weight ?: $unit->product->weight,
```

Comment:
This should be NULL if there is no weight set so that it will default to the product weight when loading. If the product weight changes, the unit weight will not change with it.